### PR TITLE
cast to double

### LINKED ISSUE #1: The resolution of the angle measurements is 1 degree.
https://github.com/sbgisen/libdynamixel/blob/280f3a32812a57b66963dd06af4c25fc1074dd52/src/dynamixel/servos/servo.hpp#L198

The above part function calculates the angle as an int type and then casts it to a double type.

### PR DESCRIPTION
**Pull request by jsupratman13**
resolves #1

### CODE REVIEW

**Review by nyxrobotics — APPROVED** [review]

File: src/dynamixel/servos/servo.hpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ namespace dynamixel {
             {
                 typename Servo<Model>::ct_t::present_position_t pos;
                 Servo<Model>::protocol_t::unpack_data(st.parameters(), pos);
-                double deg = ((pos - ct_t::min_goal_position) * (ct_t::max_goal_angle_deg - ct_t::min_goal_angle_deg) / (ct_t::max_goal_position - ct_t::min_goal_position)) + ct_t::min_goal_angle_deg;
+                double deg = (((double)pos - (double)ct_t::min_goal_position) * ((double)ct_t::max_goal_angle_deg - (double)ct_t::min_goal_angle_deg) / ((double)ct_t::max_goal_position - (double)ct_t::min_goal_position)) + (double)ct_t::min_goal_angle_deg;
```
Comment (ssr-yuki) [inline src/dynamixel/servos/servo.hpp:198]: You should use `static_cast` here.

Comment (jsupratman13) [reply src/dynamixel/servos/servo.hpp:198]: seems like a bit overkill but why not? better be safe

Comment (jsupratman13) [reply src/dynamixel/servos/servo.hpp:198]: will from
```
double deg = (((double)pos - (double)ct_t::min_goal_position) * ((double)ct_t::max_goal_angle_deg - (double)ct_t::min_goal_angle_deg) / ((double)ct_t::max_goal_position - (double)ct_t::min_goal_position)) + (double)ct_t::min_goal_angle_deg; 
```
to

```
double deg = ((static_cast<double>(pos) - static_cast<double>(ct_t::min_goal_position)) * (static_cast<double>(ct_t::max_goal_angle_deg) - static_cast<double>(ct_t::min_goal_angle_deg)) / (static_cast<double>(ct_t::max_goal_position) - static_cast<double>(ct_t::min_goal_position))) + static_cast<double>(ct_t::min_goal_angle_deg); 
```
suffice?
if so I will open another PR to apply this change

Comment (ssr-yuki) [reply src/dynamixel/servos/servo.hpp:198]: https://stackoverflow.com/questions/1609163/what-is-the-difference-between-static-cast-and-c-style-casting

Comment (ssr-yuki) [reply src/dynamixel/servos/servo.hpp:198]: > will from
> 
> ```
> double deg = (((double)pos - (double)ct_t::min_goal_position) * ((double)ct_t::max_goal_angle_deg - (double)ct_t::min_goal_angle_deg) / ((double)ct_t::max_goal_position - (double)ct_t::min_goal_position)) + (double)ct_t::min_goal_angle_deg; 
> ```
> 
> to
> 
> ```
> double deg = ((static_cast<double>(pos) - static_cast<double>(ct_t::min_goal_position)) * (static_cast<double>(ct_t::max_goal_angle_deg) - static_cast<double>(ct_t::min_goal_angle_deg)) / (static_cast<double>(ct_t::max_goal_position) - static_cast<double>(ct_t::min_goal_position))) + static_cast<double>(ct_t::min_goal_angle_deg); 
> ```
> 
> suffice?
> if so I will open another PR to apply this change

Yes, that's right. It is much safer and better implementation in C++.

Comment (jsupratman13) [reply src/dynamixel/servos/servo.hpp:198]: got it will open a PR

**Pull request merged**